src/System.Private.ServiceModel/src/System/ServiceModel/Channels/CoreClrSocketConnection.cs

```diff
@@ -550,6 +550,20 @@ protected override void WriteCore(byte[] buffer, int offset, int size, bool imme
             }
         }
 
+        protected override void TraceWriteStart(int size, bool async)
+        {
+            var remoteEndpoint = (IPEndPoint)_socket.RemoteEndPoint;
+            string remoteEndpointAddressString = remoteEndpoint.Address + ":" + remoteEndpoint.Port;
```

Comment:
This is same behavior as on Desktop. We only call Write after a socket is connected so _socket will be non-null. I'll add a Contract.Assert